beacon/goclient/proposer.go

```diff
@@ -38,16 +45,49 @@ func (gc *goClient) GetBeaconBlock(slot phase0.Slot, committeeIndex phase0.Commi
 	case spec.DataVersionAltair:
 		return beaconBlock.Altair, beaconBlock.Version, nil
 	case spec.DataVersionBellatrix:
+		if b := beaconBlock.Bellatrix.Body; b != nil && b.ExecutionPayload != nil {
+			gc.logBlock(slot, b.ExecutionPayload.BlockHash, beaconBlock.Version, false)
+		}
 		return beaconBlock.Bellatrix, beaconBlock.Version, nil
 	case spec.DataVersionCapella:
+		if b := beaconBlock.Capella.Body; b != nil && b.ExecutionPayload != nil {
+			gc.logBlock(slot, b.ExecutionPayload.BlockHash, beaconBlock.Version, false)
+		}
 		return beaconBlock.Capella, beaconBlock.Version, nil
 	default:
-		return nil, DataVersionNil, errors.New(fmt.Sprintf("beacon block version %s not supported", beaconBlock.Version))
+		return nil, DataVersionNil, fmt.Errorf("beacon block version %s not supported", beaconBlock.Version)
 	}
 }
 
-func (gc *goClient) GetBlindedBeaconBlock(slot phase0.Slot, committeeIndex phase0.CommitteeIndex, graffiti, randao []byte) (ssz.Marshaler, spec.DataVersion, error) {
-	return nil, DataVersionNil, nil
+func (gc *goClient) GetBlindedBeaconBlock(slot phase0.Slot, graffiti, randao []byte) (ssz.Marshaler, spec.DataVersion, error) {
+	sig := phase0.BLSSignature{}
+	copy(sig[:], randao[:])
+
+	reqStart := time.Now()
+	beaconBlock, err := gc.client.BlindedBeaconBlockProposal(gc.ctx, slot, sig, graffiti)
+	if err != nil {
+		return nil, 0, err
+	}
+	metricsProposerDataRequest.Observe(time.Since(reqStart).Seconds())
```

Comment:
maybe add a label to this metric about this request being for blinded block

Reply:
I don't want to add another label to `ssv_beacon_data_request_duration_seconds` because it's generic for all duty types. I can create a new `role` value for blinded blocks, but that would require fixing the dashboards. Also, we might want to think what we want to show in Grafana. Therefore, I'd extract this out of scope of this PR and define first what changes in Grafana are needed. @y0sher Do you agree?

Reply:
Agree, @Tomblox1 , lets make sure we know we need to revisit metrics especially for distinguishing blinded blocks.

Reply:
IO1-3274